### ISSUE TITLE
Replace marked by markdown-it?

### ISSUE THREAD

**Issue by tpoisot** [atom/markdown-preview#248]
The package currently uses marked to render markdown into html. I think it would be worth discussing whether this is the best option -- [markdown-it][mit] seems to be more advanced, easier to expand, and is both GFM and CommonMark compatible.

[mit]: https://github.com/markdown-it/markdown-it

**Comment by mnquintana** [comment]
Thanks for bringing this up @tpoisot! I agree, I think we should probably move markdown-preview (as well as Atom core and other packages that use marked) over to markdown-it, but we'll need more people to weigh in on this.

/cc @atom/feedback

**Comment by lee-dohm** [comment]
I looked through the demo page and while I saw lots of things talking about CommonMark and features that I know GFM supports, I didn't see a GFM mode like the one for CommonMark. Is there some easy way to configure markdown-it to enable the features that GFM supports and disable the ones it doesn't?

**Comment by mnquintana** [comment]
https://markdown-it.github.io/markdown-it/#presetname

Looks like we'd have to make a PR to add it as a preset.

**Comment by tpoisot** [comment]
@lee-dohm good question. From my reading of the docs, markdown-it is based on plugins, so GFM would be a set of plugins (most of which are already included). On a broader scale, maybe CommonMark will eventually take over, and I'd like to see some flexibility in the different tools. This is where markdown-it coud work, as in the package settings, users can select which of the plugins they want to activate?

**Comment by tpoisot** [comment]
@mnquintana Isn't what the "default" does? 

> default - similar to GFM, used when no preset name given. Enables all available rules, but still without html, typographer & autolinker.

**Comment by mnquintana** [comment]
@tpoisot I wasn't totally sure on that – similar != the same as. So I couldn't tell at a glance if it was identical to GFM.

**Comment by tpoisot** [comment]
@mnquintana right, good call. This is where the specs will be useful. The criterion would be that replacing marked by markdown-it would not break anything when the "right* config is used.

**Comment by clkao** [comment]
`markdown-it` does seem to be a more flexible implementation.  we should also allow users to configure custom plugins used by markdown-it.

**Comment by leipert** [comment]
Have a look at [Markdown Preview Plus][mpp]. It uses markdown-it. It now uses markdown-it and markdown-it-math, but our goal is to support as many of pandocs features as possible with markdown-it.

We also want to look into the extensibility by the user (maybe add hooks or something like that, so that the user can use own plugins).

You can also use pandoc instead of markdown-it, if you want.

Disclaimer: I am a maintainer of mpp.

[mpp]: https://atom.io/packages/markdown-preview-plus

**Comment by nathansobo** [comment]
> Disclaimer: I am a maintainer of mpp.

Excited to see the `-plus` process at work with markdown-preview. Best wishes.

**Comment by joallard** [comment]
It would be nice to have a parser that is kept (more) up to date with the Commonmark spec. (like backslash line-breaks I'm needing!) Sorry for the noise, but if I can give any encouragement on this, the better.

**Comment by nathansobo** [comment]
cc @BinaryMuse

**Comment by leipert** [comment]
@altimmons According to the package.json it seems to be marked:

https://github.com/atom/markdown-preview/blob/master/package.json#L16
